### PR TITLE
Add API Scanning Setup section to scanners.rst

### PR DESCRIPTION
**Pull request by Unkn0wnCat**
## Proposed change

This PR adds an "API Scanning Setup" section to docs/scanners.rst with instructions on how to connect compatible scanners directly to the Paperless-ngx API.

This change is motivated by the fact I got multiple emails / DMs asking me how to set this up. Due to that I think it's beneficial to have this section in the official documentation.

## Type of change

- [x] Other (please explain)

Documentation update.

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://paperless-ngx.readthedocs.io/en/latest/extending.html#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by Unkn0wnCat** [comment]
This mirrors jonaswinkler/paperless-ng#1691 as I was told that project was be dead in the water.

File: docs/scanners.rst [hunk 1]
```diff
@@ -112,3 +112,23 @@ On Android, you can use these applications in combination with one of the :ref:`
...
+
+API Scanning Setup
+==================
+
+This sections contains information on how to set up scanners to post directly to :ref:`Paperless API <api-file_uploads>`.
+
+Doxie Q2
+--------
+
+This part assumes your Doxie is connected to WiFi and you know it's IP.
```
Comment (qcasey) [inline docs/scanners.rst:124]:
```suggestion
This part assumes your Doxie is connected to WiFi and you know its IP.
```

File: docs/scanners.rst [hunk 2]
```diff
@@ -112,3 +112,23 @@ On Android, you can use these applications in combination with one of the :ref:`
...
+
+1. Open your Doxie web UI by navigating to its IP address
+2. Navigate to Options -> Webhook
+3. Set the *URL* to ``https://[your-paperless-ng-instance]/api/documents/post_document/``
+4. Set the *File Parameter Name* to ``document``
+5. Add the username and password to the respective fields (Consider creating a user just for your Doxie)
+6. Click *Submit* at the bottom of the page
+
+Congrats, you can now scan directly from your Doxie to your Paperless-ng instance!
```
Comment (qcasey) [inline docs/scanners.rst:133]:
```suggestion
Congrats, you can now scan directly from your Doxie to your Paperless-ngx instance!
```

File: docs/scanners.rst [hunk 3]
```diff
@@ -112,3 +112,23 @@ On Android, you can use these applications in combination with one of the :ref:`
...
+This sections contains information on how to set up scanners to post directly to :ref:`Paperless API <api-file_uploads>`.
+
+Doxie Q2
+--------
+
+This part assumes your Doxie is connected to WiFi and you know it's IP.
+
+1. Open your Doxie web UI by navigating to its IP address
+2. Navigate to Options -> Webhook
+3. Set the *URL* to ``https://[your-paperless-ng-instance]/api/documents/post_document/``
```
Comment (qcasey) [inline docs/scanners.rst:128]:
```suggestion
3. Set the *URL* to ``https://[your-paperless-ngx-instance]/api/documents/post_document/``
```

**Review by qcasey — CHANGES_REQUESTED** [review]
Some very minor corrections, but otherwise LGTM. Seems like a useful feature :)

**Comment by Unkn0wnCat** [comment]
Thanks for the fixes @qcasey, I'm really annoyed at myself for making the "it's/its" mistake lol

**Review by qcasey — APPROVED** [review]

**Review by tribut — APPROVED** [review]

**Comment by sevmonster** [comment]
> Hi, sounds interesting, but I have no clue, what Doxie is. A few words in the first line would be a nice feature. Cheers Tido

Doxie Q2 is a recommended scanner that is mentioned in the table above the added section. Doesn't seem to be particularly useful to add any more information about it, and setting up the scanner itself is probably out of scope of the article.

**Comment by Unkn0wnCat** [comment]
@sevmonster:
> setting up the scanner itself is probably out of scope of the article

Yeah it is somewhat over the scope, but that document was the best fit for those API setup instructions. And I really wanted to have those right with the recommendations as I've received many mails asking me how to do it, so I see it as a quality of life improvement for both the readers of the docs and me to have it right there.

@tido-:
I think that will not be necessary, as that section is primarily aimed at readers who have a Doxie already and just want to connect it. On the other hand interested people can just do a quick web-search to find out more about it. I don't like the idea of including a "sales-pitch"-y section right there in the docs, as I think they should be as neutral about the scanners as possible.

Have a nice day!

**Comment by tido-** [comment]
> "sales-pitch"  

I don't know where you found that in my words.
> Doxie Q2 is a recommended scanner

Thank you I had no clue, that this is a hardware brand like brother, HP or such. I thought it was about software.

**Comment by sevmonster** [comment]
> I don't know where you found that in my words.

What he means is that including specific information outside of how to set up the Paperless integration sounds rather "sales-pitch"-y, and I agree. API setup is in-scope with the list and useful to potential integrators. If you need any more info about the scanner just Google it—being partial to any one brand without reason is not a good look.

> Thank you I had no clue, that this is a hardware brand like brother, HP or such. I thought it was about software.

Why would a software product be on a recommended hardware list...? And a simple search would have shown otherwise.

**Comment by tido-** [comment]
> on a recommended hardware list.

Forgive my ignorance, I just read the new text, not where it belongs, apart from to the instructions.

@Unkn0wnCat 
I read it now again inside the manual and I think something is missing on the 5th step. (What comes before that?) As it is written for beginners, I guess.
Before the 5th step, rather the 1st step is: create a user for your scanner like: command xyz, limit its access to xyz or only in the group xyz.

**Comment by Unkn0wnCat** [comment]
@tido- 

> > on a recommended hardware list.
> 
> Forgive my ignorance, I just read the new text, not where it belongs, apart from to the instructions.

Ah yeah without the context it can be a bit confusing, no problem. 

> I read it now again inside the manual and I think something is missing on the 5th step. (What comes before that?) As it is written for beginners, I guess.
> Before the 5th step, rather the 1st step is: create a user for your scanner like: command xyz, limit its access to xyz or only in the group xyz.

I agree, though this may be something to be added somewhere else in the docs and linked to. Linking to a page explaining user creation instead of adding it to the recommendations page and (worst-case) repeating it for every scanner will help keep the page uncluttered.

**Pull request merged**